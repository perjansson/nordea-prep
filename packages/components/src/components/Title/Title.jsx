import React from 'react';
import styled from 'styled-components';

const Title = styled.h1`
    font-size: 84px;
    color: #fff;
`;

export default Title;
