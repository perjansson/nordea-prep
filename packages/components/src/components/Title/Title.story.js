import React from 'react'
import { storiesOf } from '@storybook/react'
import styled from 'styled-components'

import Title from './Title'

const Wrapper = styled.div`
    & > * {
        color: pink;
    }
`

storiesOf('Title', module)
    .addDecorator(story => <Wrapper>{story()}</Wrapper>)
    .add('with text', () => <Title>My Fancy Title!</Title>)
    .add('with emoji', () => <Title emoji="👍">My Fancy Title!</Title>)
