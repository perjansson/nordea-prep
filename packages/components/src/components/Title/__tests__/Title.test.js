import React from 'react'
import { shallow, mount } from 'enzyme'
import 'jest-styled-components'

import Title from '../Title'

describe('<Title />', () => {
    const titleText = 'title text'

    describe('enzyme mount', () => {
        it('renders text', () => {
            const wrapper = mount(<Title>{titleText}</Title>)
            expect(wrapper.text()).toEqual(titleText)
        })

        it('renders the text and emoji', () => {
            const wrapper = mount(<Title emoji="👍">{titleText}</Title>)
            expect(wrapper.text()).toEqual(`${titleText} 👍`)
        })
    })

    describe('enzyme shallow', () => {
        it('calls onClick when clicked', () => {
            const onClickFn = jest.fn()
            const component = <Title onClick={onClickFn}>{titleText}</Title>
            const wrapper = shallow(component)
            wrapper.simulate('click')
            expect(onClickFn).toHaveBeenCalled()
        })
    })

    describe('jest snapshot', () => {
        it('renders shallow component', () => {
            const wrapper = shallow(<Title emoji="👍">{titleText}</Title>)
            expect(wrapper).toMatchSnapshot()
        })
        it('renders mounted component', () => {
            const wrapper = mount(<Title emoji="👍">{titleText}</Title>)
            expect(wrapper).toMatchSnapshot()
        })
        it('checks style rules', () => {
            const wrapper = mount(<Title emoji="👍">{titleText}</Title>)
            expect(wrapper).toHaveStyleRule('font-size', '84px')
        })
    })
})
