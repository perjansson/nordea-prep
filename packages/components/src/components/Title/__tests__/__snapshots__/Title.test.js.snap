// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> jest snapshot renders mounted component 1`] = `
.c0 {
  font-size: 84px;
  margin: 20px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > * {
  font-size: 0.7em;
}

<Title
  className=""
  emoji="👍"
  onClick={[Function]}
  style={Object {}}
>
  <styled.h1
    className=""
    onClick={[Function]}
    style={Object {}}
  >
    <h1
      className="c0"
      onClick={[Function]}
      style={Object {}}
    >
      title text
      <span
        aria-label="smiley-with-star-eyes"
        role="img"
      >
         
        👍
      </span>
    </h1>
  </styled.h1>
</Title>
`;

exports[`<Title /> jest snapshot renders shallow component 1`] = `
<styled.h1
  className=""
  onClick={[Function]}
  style={Object {}}
>
  title text
  <span
    aria-label="smiley-with-star-eyes"
    role="img"
  >
     
    👍
  </span>
</styled.h1>
`;
